uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/microsoftsts/IMicrosoftStsLoginComponentHandler.java

```diff
@@ -0,0 +1,48 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.microsoftsts;
+
+import com.microsoft.identity.client.ui.automation.interaction.IOAuth2LoginComponentHandler;
+
+public interface IMicrosoftStsLoginComponentHandler extends IOAuth2LoginComponentHandler {
```

Comment:
Thoughts on taking a more event driven approach with these (and other) interface method names?

Examples:
- `onSpeedBumpShown()`
- `onEnrollmentConfirmationPageShown()`
- `onEnrollmentAccepted()`
- `onEnrollmentDeclined()`

Reply:
I do like your suggestion, however, I'll stick to the current names for now and tackle the naming in a separate PR later this month. (because I've already added more code on a local branch and I don't want to deal with many merge conflicts right now 😁)

Reply:
Sounds good